Tweetinvi.Controllers/Tweet/TweetQueryGenerator.cs

```diff
@@ -120,21 +120,38 @@ public string GetPublishRetweetQuery(long tweetId)
         }
 
         // Get Retweets
-        public string GetRetweetsQuery(ITweetDTO tweetDTO)
+        public string GetRetweetsQuery(ITweetDTO tweetDTO, int maxRetweetsToRetrieve)
         {
             if (!_tweetQueryValidator.IsTweetPublished(tweetDTO))
             {
                 return null;
             }
 
-            return GetRetweetsQuery(tweetDTO.Id);
+            return GetRetweetsQuery(tweetDTO.Id, maxRetweetsToRetrieve);
         }
 
-        public string GetRetweetsQuery(long tweetId)
+        public string GetRetweetsQuery(long tweetId, int maxRetweetsToRetrieve)
         {
-            return string.Format(Resources.Tweet_Retweet_GetRetweets, tweetId);
+            return string.Format(Resources.Tweet_Retweet_GetRetweets, tweetId, maxRetweetsToRetrieve);
         }
-        
+
+        #region Get Retweeter Ids
+        public string GetRetweeterIdsQuery(ITweetIdentifier tweetIdentifier, bool stringifyIds)
+        {
+            if (!_tweetQueryValidator.IsValidTweetIdentifier(tweetIdentifier))
+            {
+                return null;

```

Comment:
This will need to throw an `ArgumentException`. In the past we returned null but this will be changed as part of 0.9.13.0. All the invalid arguments will now throw an `ArgumentException`. Issue #196